backend/src/graphql.ts

```diff
@@ -117,114 +132,123 @@ schema.objectType({
 const EditorNode = schema.inputObjectType({
   name: 'EditorNode',
   definition(t) {
-    t.string('type', { nullable: true }),
-      t.string('text', { nullable: true }),
-      t.boolean('italic', { nullable: true }),
-      t.boolean('bold', { nullable: true }),
-      t.field('children', {
-        type: EditorNode,
-        list: true,
-        nullable: true,
-      })
+    t.string('type', { nullable: true })
+    t.string('text', { nullable: true })
+    t.boolean('italic', { nullable: true })
+    t.boolean('bold', { nullable: true })
+    t.field('children', {
+      type: EditorNode,
+      list: true,
+      nullable: true,
+    })
   },
 })
 
 schema.queryType({
   definition(t) {
     t.list.field('posts', {
       type: 'Post',
-      resolve: async (parent, args, ctx) => ctx.db.post.findMany(),
-    }),
-      t.field('postById', {
-        type: 'Post',
-        args: {
-          id: intArg(),
-        },
-        resolve: async (parent, args, ctx) => {
-          return await ctx.db.post.findOne({
-            where: {
-              id: args.id,
+      args: {
+        authorId: intArg({ required: true }),
+      },
+      resolve: async (_root, args, ctx) => {
+        return ctx.db.post.findMany({
+          where: {
+            author: { id: args.authorId },
+          },
+        })
+      },
+    })
```

Comment:
Lines 151-here are new though and aren't due to the prettier formatting changes

Reply:
I wonder if the `authorId` should be optional here 🤔 My thought earlier was more to extend this query to be able to serve this use case. What do you think?

Reply:
Since we already have a `feed` query that finds posts based on things other than `authorId`, I was trying to think of a use case when we'd want to use the `posts` query for purposes other than a feed and not have an `authorId`, but I couldn't come up with one. My posts, drafts, and the profile page all require an `authorId`. Perhaps it should be called `postByAuthorId` like we have `postById`? I'm open to suggestions! 😄